lib/netext/grpcext/conn.go

```diff
@@ -196,37 +196,36 @@ func (statsHandler) TagRPC(ctx context.Context, _ *grpcstats.RPCTagInfo) context
 // HandleRPC implements the grpcstats.Handler interface
 func (h statsHandler) HandleRPC(ctx context.Context, stat grpcstats.RPCStats) {
 	state := h.vu.State()
-	tags := getTags(ctx)
-
-	// If the request is done by the reflection handler
-	// then the tags will be nil.
-	//
-	// In this case, we can reuse the State's Tags.
-	if tags == nil {
-		tags = state.Tags.BranchOut()
+	stateRPC := getRPCState(ctx) //nolint:ifshort
+
+	// If the request is done by the reflection handler then the tags will be
+	// nil. In this case, we can reuse the VU.State's Tags.
+	if stateRPC == nil {
+		// TODO: investigate this more, there has to be a way to fix it :/
+		stateRPC = &rpcState{tags: state.Tags.GetCurrentValues()}
 	}
 
 	switch s := stat.(type) {
 	case *grpcstats.OutHeader:
+		// TODO: figure out something better, e.g. via TagConn() or TagRPC()?
 		if state.Options.SystemTags.Has(metrics.TagIP) && s.RemoteAddr != nil {
+			stateRPC.ip = s.RemoteAddr.String()
```

Comment:
Is this used?

Reply:
No, but my original plan was to actually add the `ip` tag in the `case *grpcstats.End:` below, that's why I added it :man_facepalming: This whole thing probably needs to be refactored, it seems very fragile, and that `stateRPC == nil` check is very suspect... :confused: In some other PR though, for now I'll remove the `stateRPC.ip`

Reply:
Resolved by https://github.com/grafana/k6/pull/2638/commits/eec1bc9afb712ba699034f6d27a5eacb56e21017